pkg/dashboard/ui/src/app/components/data-wrappers/create-function-data-wrapper/create-function-data-wrapper.component.js

```diff
@@ -54,13 +55,32 @@
             return NuclioFunctionsDataService.getTemplates()
                 .then(function (templates) {
                     lodash.forIn(templates, function (value) {
-                        var title = value.metadata.name.split(':')[0] + ' (' + value.spec.runtime + ')';
+                        if (!lodash.has(value, 'rendered') && lodash.has(value, 'template')) {
+                            var template = YAML.parse(value.template.replace(/{{\s.+\s}}/g, '"$&"'));
+
+                            lodash.set(value, 'rendered.spec', template.spec);
+                        }
+
+                        if (!lodash.has(value, 'rendered.metadata.name')) {
+                            lodash.set(value, 'rendered.metadata.name', lodash.get(value, 'metadata.name', 'default'));
```

Comment:
I will remove `default` from `lodash.get(value, 'metadata.name', 'default')` before merge. It's just for testing, because one template come with empty `metadata`